### DISCUSSION TITLE
Use case for "Enable remote index downloads"

### DISCUSSION THREAD

**Discussion by goyalyashpal** [apache/netbeans#6683] [Q&A]
Index < Maven < Java <-- Options < Tools < Menu bar

* What happens if i disable this option of "Enable remote index downloads"

* What are the situations where one would like to enable this?
Considering that this downloads from [repo.maven] which is a WHOPPING 2.15 GiB

* How does it differ from this feature outlined in [OpenBeans] "Online Maven support" feature (Quoted below)
<br/>“
&nbsp;&nbsp;&nbsp;&nbsp; _**Java: Online Maven support.**_
&nbsp;&nbsp;&nbsp;&nbsp; _Maven queries will be sent to Maven Central instead of downloading the Maven Central cache which takes GBs of your disk._
„

[repo.maven]: https://repo.maven.apache.org/maven2/.index/nexus-maven-repository-index.gz
[openbeans]: http://www.openbeans.org/comparison.html

### Screenshots:

* <img src="https://github.com/apache/netbeans/assets/19423063/35630793-0ab6-44ed-b912-0e586b826738" width=350px title="repo .index .gz download size from browser" />

* <img src="https://github.com/apache/netbeans/assets/19423063/84f744f7-9245-4758-bcaf-93ee0ef5bd79" width=650px title="index download progress in netbeans" />

* <img src="https://github.com/apache/netbeans/assets/19423063/bfadca25-af67-4cca-8f81-601ae4ec3ead" width=550px title="netbeans options" />

**Comment by mbien** [comment]
> What happens if i disable this option of "Enable remote index downloads"

When remote index downloads are disabled NB will only scan your local `.m2` repository. There are some exceptions, for example if you search for a unknown class name, this would optionally use both local .m2 search and the SMO service, screenshots: #5984
Or if you try to find javadoc for a jar, it would run a remote SH1 query.

If you don't use maven much (esp not with maven poms) and/or have storage problems, you can keep remote index downloads disabled. NetBeans won't download anything by default The remote index of maven central is large since it contains information about every java artifact ever released but it is updated incrementally. Updates are usually about 10 MB deltas and run fast. The last screenshot you posted describes what it is used for.

If you want to reduce the footprint a bit of the index files you can also enable the filter. E.g if you set it to 2 years it will drop any information older than 2 years while processing the remote index file. This should be ideally enabled before first download.

But if you don't need that at all you can leave it disabled. Without remote index, NB will for example only suggest artifact version updates in poms if they happen to be already in your local repository, completion will be also limited to what NB already knows about.
